pkg/syncer/storagepool/storageclasswatch.go

```diff
@@ -399,24 +410,26 @@ func isInvalidProfileErr(ctx context.Context, err error) (bool, []string) {
 
 // Query SPBM to get the list of policies that the given list of datastores
 // is compatible with. Returns a dict of format:
-// datastoreMoId -> []string of policyIds
+// datastoreMoId -> map of policy IDs
 //
-// If storage classes haven't recently changed, and if forceRefreshforceRefresh
+// If storage classes haven't recently changed, and if forceRefresh
 // is false, will return cached data.
 func (w *StorageClassWatch) getDatastoreToPolicyCompatibility(ctx context.Context,
-	datastores []*cnsvsphere.DatastoreInfo, forceRefresh bool) (map[string][]string, error) {
+	datastores []*cnsvsphere.DatastoreInfo, forceRefresh bool) (map[string]map[string]bool, error) {
 	log := logger.GetLogger(ctx)
 	if !forceRefresh {
 		return w.dsPolicyCompatMapCache, nil
 	}
-	datastoreToPoliciesMap := make(map[string][]string)
+	datastoreToPoliciesMap := make(map[string]map[string]bool)
```

Comment:
what is the reason behind chaging list to map?

Reply:
The reason why I added this was when I started working on this, I realised that there were duplicate entries in the `dataStore -> Policy` list. This was because of the previous impl of `refreshStorageClassCache` where we were not removing duplicates for storage policies that could have multiple SCs[WFFC and immediate for instance]. Since I already am ensuring that there won't be duplicate policy IDs in the newer impl, this can be reverted.

Reply:
I've reverted back this change.